### ISSUE TITLE
Saving to monochrome AVIF does not seem to work.

### ISSUE THREAD

**Issue by therahedwig** [strukturag/libheif#447]
This worked in the past, but right now, using libheif to encode gray data into avif, or the example heif_enc program to convert a grayscale png to avif results in the data being encoded as RGBA.

Opening grayscale avifs works fine, and saving and loading a heif file also works fine.

I build master just half an hour ago to make sure I wasn't missing fixes, but the problem remains.

EDIT: My colleague pointed out rav1e doesn't support monochrome, but I am using libaom. I think for rave1, 'converting' to YCbCr is acceptable, but if libheif cannot by itself tell you that only the Y channel is used, it should be documented that this might be the caste.

**Comment by farindk** [comment]
I've changed the AOM encoder plugin to use monochrome encoding for greyscale input instead of YCbCr 4:2:0 fallback.
Please check.

**Comment by novomesk** [comment]
I remember that libavif enabled grayscale only when libaom was newer than 2.0.0
There was some bug in libaom 2.0.0 and libavif saved grayscale as yuv420 to avoid the bug.
I recommend to test with libaom 2.0.0 and with new 2.0.2 (or 2.1.0-rc1). I think many systems still use older version.

**Comment by therahedwig** [comment]
> I've changed the AOM encoder plugin to use monochrome encoding for greyscale input instead of YCbCr 4:2:0 fallback.
> Please check.

Yes, this seems to work! As for Novomesk's comments... I am using libaom 1.0 here. I'm okay with yuv420 being used, but I do need to see this peculiarity documented because I'd need to do some weird stuff to ensure that the color profile loading doesn't go haywire. Like, right now it's sounding like I would need to double check that the profile is a grayscale profile before loading it and then adjust the pixel decoding parts to treat the pixels as grayscale instead of trusting libheif to tell me it's grayscale.

Thanks a lot!

**Comment by novomesk** [comment]
@therahedwig encoding with libaom 1.0.0 is very slow. I recommend to use 2.0.2

**Comment by therahedwig** [comment]
Ok, so this works for libaom. For rav1e it goes to YCrCb, which Chrome doesn't understand, so I've just added a checkbox for monochrome images that says 'convert to sRGB for extra compatibility' which people can use if they are using rav1e but really need chrome to understand it. On loading, we double check which color model the color profile has and then use that to guide to us through decoding.

Everything got merged just now, so Krita 5.0 will release with updated heif and pretty good avif support.
